enterprise/api/v1/views/enterprise_customer_sso_configuration.py

```diff
@@ -59,3 +78,115 @@ def oauth_orchestration_complete(self, request, configuration_uuid, *args, **kwa
+        # or are a staff member
+        config_customer = sso_configuration_record.enterprise_customer
+        if not check_user_part_of_customer(request.user, config_customer):
+            return Response(status=HTTP_404_NOT_FOUND)
+
+        serializer = self.serializer_class(sso_configuration_record)
+        return Response(serializer.data)
+
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, *args, **kwargs: get_customer_from_request(request)
+    )
+    def list(self, request, *args, **kwargs):
+        list_queryset = self.queryset
+
+        # If the request includes a customer uuid, we require through permissioning that the requesting user is an
+        # admin of that customer
+        if provided_customer := request.query_params.get('enterprise_customer'):
+            enterprise_customer = EnterpriseCustomer.objects.filter(pk=provided_customer)
+            list_queryset = list_queryset.filter(enterprise_customer__in=enterprise_customer)
+
+        # For non-staff users, only show the configurations that are connected to enterprise customers
+        # that they are associated with
+        if not request.user.is_staff:
+            ent_customers = EnterpriseCustomer.objects.filter(
+                enterprise_customer_users__in=EnterpriseCustomerUser.objects.filter(user_id=request.user.id)
+            )
+            list_queryset = list_queryset.filter(enterprise_customer__in=ent_customers)
+
+        serializer = self.serializer_class(list_queryset, many=True)
+        return Response(serializer.data, status=HTTP_200_OK)
+
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, *args, **kwargs: get_customer_from_request(request)
+    )
+    def create(self, request, *args, **kwargs):
+        # Force the enterprise customer to be the one associated with the user
+        request_data = request.data.dict().copy()
+        requesting_user_customer = request_data.get('enterprise_customer')
+        if requesting_user_customer:
+            try:
+                enterprise_customer = EnterpriseCustomer.objects.get(uuid=requesting_user_customer)
+            except EnterpriseCustomer.DoesNotExist:
+                return Response({'error': 'Must provide valid enterprise customer'}, status=HTTP_400_BAD_REQUEST)
+            if not check_user_part_of_customer(request.user, enterprise_customer):
+                return Response({'error': 'Must provide valid enterprise customer'}, status=HTTP_400_BAD_REQUEST)
+            request_data['enterprise_customer'] = enterprise_customer
+        else:
+            return Response({'error': 'Must provide valid enterprise customer'}, status=HTTP_400_BAD_REQUEST)
+        try:
+            new_record = EnterpriseCustomerSsoConfiguration.objects.create(**request_data)
+        except TypeError as e:
+            LOGGER.error(f'Error creating SSO configuration record: {e}')
```

Comment:
Same error text used twice

Reply:
one is a logging message the other is the response returned to the user

Reply:
Sure but we can keep the error string in a single place.